apps/ledger-live-mobile/src/screens/NpsRatingsModal/Form.tsx

```diff
@@ -0,0 +1,128 @@
+
+const injectedJavascript = `
+const ratingOptions = document.querySelectorAll('div[data-qa="nps-opinion-scale-step"]');
+
+ratingOptions.forEach((option) => {
+  option.addEventListener('click', () => {
+    const rate = option.getAttribute('data-value-number');
+    window.ReactNativeWebView.postMessage('nps-submit-' + rate);
+  });
+});
+
+document.addEventListener('click', (event) => {
+  const target = event.target;
+  if (target.matches('button[data-qa*="submit-button"]')) {
+    window.ReactNativeWebView.postMessage('form-submit');
+  }
+});
+
+true;
+`;
+
+const StyledWebview = styled(WebView)`
+  background-color: transparent; // avoids white background before page loads
+  flex: 1;
+`;
+
+type Props = {
+  setStep: (t: string) => void;
+};
+
+const Form = ({ setStep }: Props) => {
+  const { ratingsHappyMoment, ratingsFeatureParams, updateNpsRating } = useNpsRatings();
+  const language = useSelector(languageSelector);
+  const devices = useSelector(knownDevicesSelector);
+  const lastDevice = useSelector(lastSeenDeviceSelector) || devices[devices.length - 1];
+  const [selectedRate, setSelectedRate] = useState();
+
+  const notifications = useSelector(notificationsSelector);
+  const notificationsAllowed = notifications.areNotificationsAllowed;
+  const notificationsBlacklisted = Object.entries(notifications)
+    .filter(([key, value]) => key !== "areNotificationsAllowed" && value === false)
+    .map(([key]) => key)
+    .join(",");
+
+  const onMessage = useCallback(
+    event => {
+      const { data } = event.nativeEvent;
+      if (data.startsWith("nps-submit")) {
+        setSelectedRate(data.split("-")[2]);
+      }
+      if (data === "form-submit") {
+        updateNpsRating(selectedRate);
+        if (!selectedRate || selectedRate <= 7) {
+          setStep("disappointedDone");
+          screen(
+            "",
+            "NPS Step 2 not Happy",
+            { page: "NPS Step 2 not Happy", source: ratingsHappyMoment?.route_name },
+            true,
+          );
+        } else {
+          setStep("enjoy");
+          screen(
+            "",
+            "NPS Step 2 Happy",
+            { page: "NPS Step 2 Happy", source: ratingsHappyMoment?.route_name },
+            true,
+          );
+        }
+      }
+    },
+    [ratingsHappyMoment?.route_name, selectedRate, setStep, updateNpsRating],
+  );
+  const formUrlSplitted = ratingsFeatureParams?.typeform_url.split("?");
+  const formUrl =
+    formUrlSplitted[0] +
+    `#app_version=${appVersion}` +
+    `&app_language=${language}` +
+    `&platform_os=${Platform.OS}` +
+    `&platform_version=${Platform.Version}` +
+    `&model_id=${lastDevice?.modelId}` +
+    `&firmware_version=${lastDevice?.deviceInfo?.version}` +
+    `&notifications_allowed=${notificationsAllowed}` +
+    `&notifications_blacklisted=${notificationsBlacklisted}` +
+    `&done?${formUrlSplitted[1] || ""}`;
```

Comment:
better to use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) for this type of URL manipulation, rather than direct string manipulation which is more error prone